### PR TITLE
#1767 Add validate unit test

### PR DESCRIPTION
**Pull request by DavidMi-1**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1767)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by konekoya** [comment]
@DavidMi-1 Awesome! Please be reminded that:
- Add a `#` symbol at the beginning of your PR title
- Capitalize the first letter of the title which in your case __add__

Thanks :)

File: ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import validate from '../validate';
+import * as generate from 'utils/generate';
+import { cleanup } from '@testing-library/react';
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js]: We usual do:

```js
// Third-party libraries usually are listed at the very top of the file
import { cleanup } from '@testing-library/react';
// Add a new line here so it's easier to distinguish them
// User defined modules are listed here
import validate from '../validate';
import * as generate from 'utils/generate';
```

**Comment by DavidMi-1** [comment]
> @DavidMi-1 Awesome! Please be reminded that:
> 
> * Add a `#` symbol at the beginning of your PR title
> * Capitalize the first letter of the title which in your case **add**
> 
> Thanks :)

OK! got it.

File: ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+const values = {
+  name: generate.name(),
+  port: generate.port(),
+  user: generate.userName(),
+  password: generate.password(),
+};
+
+const expected = undefined;
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js]: Why this declaration? 🤔

File: ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+afterEach(cleanup);
+
+describe('validate()', () => {
+  it('handles success function call', async () => {
+    const result = validate(values);
+
+    expect(result.name).toBe(expected);
+    expect(result.port).toBe(expected);
+    expect(result.user).toBe(expected);
+    expect(result.password).toBe(expected);
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js]: You can assert that the returned `error` object is an empty object here :)

Comment (konekoya) [reply ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js]: Maybe something like: `expect(isEmpty(validate(values))).toBe(true)`. The `isEmpty` is a function coming from `lodash`

File: ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  });
+
+  it('validate name is empty', async () => {
+    values.name = '';
+    const result = validate(values);
+
+    expect(result.name).toBe('Required field');
+    expect(result.port).toBe(expected);
+    expect(result.user).toBe(expected);
+    expect(result.password).toBe(expected);
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js]: No need to check every field here, checking length should be succinct

File: ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  it('validate port is empty', async () => {
+    values.name = generate.name();
+    values.port = '';
+    const result = validate(values);
+
+    expect(result.name).toBe(expected);
+    expect(result.port).toBe('Required field');
+    expect(result.user).toBe(expected);
+    expect(result.password).toBe(expected);
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/__tests__/validate.test.js]: Same here and the rest of the tests

**Comment by konekoya** [comment]
@DavidMi-1 I've simplified the test a bit :)

**Pull request merged**